### PR TITLE
Improved docs

### PR DESCRIPTION
**Pull request by j0ack**
**Documentation**

- Fix some typos
- Create some subtitles in `porting-guide`
- Rewrite `Handles vs PyObject *` introduction in `api`

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 # Add any Sphinx extension module names here, as strings. They can be
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
-extensions = []
+extensions = ["sphinx.ext.autosectionlabel"]
```
Comment (hodgestar) [inline docs/conf.py:33]: Thanks, this looks useful!

File: docs/overview.rst [hunk 2]
```diff
@@ -57,8 +57,8 @@ Hide internal details
   the ones used by CPython.  In particular, **reference counting is not part
   of the API**: we want a more generic way of managing resources which is
   possible to implement with different strategies, including the existing
-  reference counting and/or with a moving GC (like the ones used by PyPy or
-  Java, for example).
+  reference counting and/or with a moving *Garbage Collector* (like the ones u
+  sed by PyPy or Java, for example).
```
Comment (hodgestar) [inline docs/overview.rst:61]: `used` seems to have been split over two lines as `u` and `sed`.

File: docs/overview.rst [hunk 3]
```diff
@@ -183,12 +183,12 @@ C extensions
...
 
 If you are writing a Python extension in C, you are a consumer of the HPy
-API. There are two big advantages in using HPy instead of the old Python/C
+API. There are three big advantages in using HPy instead of the old Python/C
 API:
 
   - Speed on PyPy and other alternative implementations: according to early
     :ref:`benchmarks`, an extension written in HPy can be ~3x faster than the
-    equivalent extenson writting in Python/C.
+    equivalent extension writting in Python/C.
```
Comment (hodgestar) [inline docs/overview.rst:191]:
```suggestion
    equivalent extension written in Python/C.
```

File: docs/porting-guide.rst [hunk 4]
```diff
@@ -1,23 +1,35 @@
...
 
-xxx
----------------------
+PyModule_AddObject
+------------------
 
-`PyModule_AddObject()`: replace with a regular `HPy_SetAttr_s()`.  There is no `HPyModule_AddObject()` because it has an unusual refcount behaviour (stealing a reference but only when it returns 0).
+``PyModule_AddObject()`` is replaced with a regular ``HPy_SetAttr_s()``. There
+is no ``HPyModule_AddObject()`` because it has an unusual refcount behaviour (
+stealing a reference but only when it returns 0).
```
Comment (hodgestar) [inline docs/porting-guide.rst:9]: `(stealing` was split into `( stealing`.

**Review by hodgestar — APPROVED** [review]

File: docs/porting-guide.rst (hunk 4 above)
Comment (antocuni) [reply docs/porting-guide.rst:9]: shouldn't it be `"PyModule_AddObject is replace **by** a regular HPy_SetAttr_s?

**Review by antocuni — APPROVED** [review]
apart the few minor typos already caught by @hodgestar, the PR looks good to me.
Thank you!

**Comment by hodgestar** [comment]
Merged and will tidy up the small typos now.

**Comment by hodgestar** [comment]
Minor typos fixed in 43ad913.

**Comment by j0ack** [comment]
Thanks for the review

**Pull request merged**